File: Semester 2/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Jan_22/lvl2_hw_sol/lvl2_systests.py
import sys, bisect
input = sys.stdin.readline
bat = [ ]
for i in range(int(input())):
    bat.append(list(map(int, input().split())))
tests = []
for i in range(int(input())):
    tests.append(int(input()))
tests.sort()
ans = 0
for e in bat:
    pos = bisect.bisect_left(tests, e[0])
    if pos == len(tests) or tests[pos] > e[1]:
        ans += e[2]
print(ans)

File: Semester 2/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Jan_22/lvl2_hw_sol/lvl2_systests.cpp
#include <bits/stdc++.h>
using namespace std;
const int MM = 1e5+2;
int B, F, a[MM], b[MM], val[MM], c[MM], ans;
int main(){
    ios::sync_with_stdio(0); cin.tie(0);
    cin >> B;
    for(int i=0; i<B; i++) cin >> a[i] >> b[i] >> val[i];
    cin >> F;
    for(int i=0; i<F; i++) cin >> c[i];
    sort(c, c+F);
    for(int i=0; i<B; i++) {
        int pos = lower_bound(c, c+F, a[i]) - c;
        if(pos == F || c[pos] > b[i]) ans += val[i];
    }
    cout << ans << endl;
}